### ISSUE TITLE
Training Loop Question

### ISSUE THREAD

**Issue by KSGulin** [signals-dev/Orion#218]
* Orion version: 0.1.6
* Python version: 3.6.8
* Operating System: Windows 10

### Description

I'm trying to better understand the part of tadgan.py in the snippet below. It seems that the two critics gets trained on partitions (default = 5) of the minibatch, but then the encoder and generator only get trained on the 5th partition of the data. I may be missing something, but it seems that the encoder and generator never see 4/5 of the data this way. Is that correct? 

```
for j in range(self.iterations_critic):
    x = minibatch[j * self.batch_size: (j + 1) * self.batch_size]
    z = np.random.normal(size=(self.batch_size, self.latent_dim, 1))
    epoch_cx_loss.append(
        self.critic_x_model.train_on_batch([x, z], [valid, fake, delta]))
    epoch_cz_loss.append(
        self.critic_z_model.train_on_batch([x, z], [valid, fake, delta]))

epoch_g_loss.append(
    self.encoder_generator_model.train_on_batch([x, z], [valid, valid, x]))

```

**Comment by sarahmish** [comment]
Hi @KSGulin, good question. The training approach is inspired by the algorithm proposed in [Wasserstein GAN](https://arxiv.org/abs/1701.07875). Because we shuffle the data in each epoch, batch content will change in each training loop.
If the number of epochs is large enough, we can assume that the encoder/generator will see all the data.
